### ISSUE TITLE
Add ECS Module to Terraform

### ISSUE THREAD

**Issue by yourbuddyconner** [CodaProtocol/coda-automation#42]
We should have an easy-to-deploy cluster consisting of many Docker containers. This should be our default deployment story and we should deprecate the current method of installing .deb packages on EC2 instances. 

Ideally we could also get rid of the Ansible dependency and allow the Daemon's to either self-discover one another or load their peers from a file generated at deploy-time by terraform. 

The Docker images for the Coda Daemon already exist, so it's just a matter of writing the Terraform module(s). 

This docker-based deployment model can then be modified only slightly to deploy to other cloud providers.

**Comment by yourbuddyconner** [comment]
Routing Questions: 
- Since containers will be running on an internal network, what will port forwarding look like? 
- Do we only launch one Daemon per ECS Instance, and have each ECS instance have a public IP (a waste) 
- Do we configure some kind of NAT hole-punching that will work in this context AND in the SOHO environment?

**Comment by yourbuddyconner** [comment]
My feeling here is that we have two options: 
1. Assign each Daemon container a unique Elastic IP (waste of IPs) 
2. Assign each ECS Instance a public IP, assign each daemon container a unique set of ports
  - Requires some kind of peer discovery mechanism

**Comment by yourbuddyconner** [comment]
@o1pranay let me mind dump this problem on him and the solutions above worked themselves out again. It seems like we can definitely support a nice, simple, ECS workload by assigning Public IPs to each of the ECS Instances hosting Coda Daemons.
![IMG_1352](https://user-images.githubusercontent.com/2635475/60302114-e2b6aa00-98e7-11e9-856c-234dadc911cc.JPG)

**Comment by jdbohrman** [comment]
Hey @yourbuddyconner would you like any help on this? I could spin up a rough-draft module this week if you'd like.

**Comment by yourbuddyconner** [comment]
@jdbohrman Thanks so much for your interest! This repository is still in its early stages and I have yet to put together the contribution guidelines for the infrastructure side of things. 

That said, I am happy to entertain contributions on this particular task! I only ask that you bare with me as I flesh out the "rules" as we go along. I've also got some rough beginnings of this module that I was playing with this week I can push up into a branch for you to look at. 

As far as the high-level requirements here, we need the following: 
- VPC for the ECS Cluster
- Private/Public Subnets
- Two ASGs
  - Public Nodes - Each node gets assigned Elastic IP Address
  - Private Nodes - Only addressable from within VPC
- A Coda Daemon service that chooses a random port on the host and binds the P2P port to it.
- a README detailing the module (See [this repo](https://github.com/terraform-aws-modules/terraform-aws-vpc) for an example) 

Extra Credit: 
- Nodes have dynamically configured hostnames based off a R53 zone

If you want to chat about this more, we have a discord server [here](https://discordapp.com/invite/ShKhA7J) you can find me with `@conner`. Thanks again for your interest!

**Comment by yourbuddyconner** [comment]
As an intermediate step to this, I have been playing with a couple manually-deployed ECS clusters to some success. Having`ecs-cli` being compatible with docker-compose as a crutch has been very useful for quick iteration.

**Comment by yourbuddyconner** [comment]
Just merged a PR implementing this!
